### ISSUE TITLE
Invalid JSON generated with cyclic reference

### ISSUE THREAD

**Issue by MrChoclate** [json-schema-faker/json-schema-faker#345]
JSON Schema

```
{
  "title": "cyclic - DefaultCyclic",
  "type": "object",
  "definitions": {
    "DefaultCyclic_106": {
      "type": "object",
      "properties": {
        "id": {
          "type": "integer"
        },
        "me": {
          "type": "object",
          "$ref": "#/definitions/DefaultCyclic_106"
        }
      },
      "additionalProperties": false,
      "required": [
        "id"
      ]
    }
  },
  "properties": {
    "id": {
      "type": "integer"
    },
    "me": {
      "type": "object",
      "$ref": "#/definitions/DefaultCyclic_106"
    }
  },
  "required": [
    "id"
  ],
  "additionalProperties": false
}
```
On http://json-schema-faker.js.org/ I got Buffer is not defined
And with node v7.10.1 (`jsf(schema);`) I sometimes got invalid json
```
{
  "id": 96316936,
  "me": {
    "id": 90121185,
    "me": {
      "id": 80291803,
      "me": {
        "id": 25486302,
        "me": {
          "id": 12801627,
          "me": {}
        }
      }
    }
  }
}
``` 
I used `json-schema-faker@0.5.0-rc11`

**Comment by pateketrueke** [comment]
Cool, on the browser it's totally broken (on that specific case, which is not rare), I think we need to re-plan how to build our distributables, again...

On nodejs I got results like yours, but why they're invalid?

**Comment by MrChoclate** [comment]
Because the last "me" has no key "id". It should either have his required
key "id" or be removed (since it is optional)

I used this https://github.com/ruby-json-schema/json-schema to validate the
json

On 13 Nov 2017 7:19 p.m., "Alvaro Cabrera Durán" <notifications@github.com>
wrote:

Cool, on the browser it's totally broken (on that specific case, which is
not rare), I think we need to re-plan how to build our distributables,
again...

On nodejs I got results like yours, but why they're invalid?

—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub
<https://github.com/json-schema-faker/json-schema-faker/issues/345#issuecomment-344009513>,
or mute the thread
